packages/web3/src/contract/deployment.test.ts

```diff
@@ -0,0 +1,169 @@
+      expect(typeof deployResult.issueTokenAmount).toBe('string')
+    })
+  })
+
+  describe('RunScriptResult', () => {
+    it('should validate RunScriptResult structure', () => {
+      const runScriptResult = {
+        // Base ExecutionResult fields
+        txId: '1234567890abcdef',
+        unsignedTx: 'unsignedTxHex',
+        signature: 'signatureHex',
+        gasAmount: 100,
+        gasPrice: '1000000000',
+        blockHash: 'blockHashHex',
+        codeHash: 'codeHashHex',
+        // RunScriptResult specific field
+        groupIndex: 0
+      }
+
+      expect(runScriptResult).toHaveProperty('groupIndex')
+      expect(typeof runScriptResult.groupIndex).toBe('number')
+      expect(runScriptResult.groupIndex).toBeGreaterThanOrEqual(0)
+    })
+
+    it('should validate groupIndex constraints', () => {
+      const validRunScriptResult = {
+        txId: '1234567890abcdef',
+        unsignedTx: 'unsignedTxHex',
+        signature: 'signatureHex',
+        gasAmount: 100,
+        gasPrice: '1000000000',
+        blockHash: 'blockHashHex',
+        codeHash: 'codeHashHex',
+        groupIndex: 3
+      }
+
+      expect(validRunScriptResult.groupIndex).toBeGreaterThanOrEqual(0)
+      expect(validRunScriptResult.groupIndex).toBeLessThan(4) // Assuming max groups is 4
+    })
+  })
+
+  describe('Type Compatibility', () => {
+    it('should ensure DeployContractExecutionResult extends ExecutionResult', () => {
+      const deployResult = {
+        txId: '1234567890abcdef',
+        unsignedTx: 'unsignedTxHex',
+        signature: 'signatureHex',
+        gasAmount: 100,
+        gasPrice: '1000000000',
+        blockHash: 'blockHashHex',
+        codeHash: 'codeHashHex',
+        contractInstance: {
+          address: '1234567890abcdef',
+          contractId: 'contractId'
+        }
+      }
+
+      // Test that it has all ExecutionResult properties
+      const executionResultKeys = ['txId', 'unsignedTx', 'signature', 'gasAmount', 'gasPrice', 'blockHash', 'codeHash']
+
+      executionResultKeys.forEach((key) => {
+        expect(deployResult).toHaveProperty(key)
+      })
+    })
+
+    it('should ensure RunScriptResult extends ExecutionResult', () => {
+      const runScriptResult = {
+        txId: '1234567890abcdef',
+        unsignedTx: 'unsignedTxHex',
+        signature: 'signatureHex',
+        gasAmount: 100,
+        gasPrice: '1000000000',
+        blockHash: 'blockHashHex',
+        codeHash: 'codeHashHex',
+        groupIndex: 0
+      }
+
+      // Test that it has all ExecutionResult properties
+      const executionResultKeys = ['txId', 'unsignedTx', 'signature', 'gasAmount', 'gasPrice', 'blockHash', 'codeHash']
+
+      executionResultKeys.forEach((key) => {
+        expect(runScriptResult).toHaveProperty(key)
+      })
+
+      // Test RunScriptResult specific property
+      expect(runScriptResult).toHaveProperty('groupIndex')
+    })
+  })
+})
```

Comment:
All these tests do appear to be testing basic JavaScript/TypeScript behavior rather than any meaningful application-specific logic. Do we really need it ?

Reply:
hey @pragmaxim While these tests may appear to be checking basic JavaScript/TypeScript behavior, they actually serve several crucial purposes in maintaining the contract system's type safety and reliability

but i can remove it if you like :)

Reply:
If the results were returned from the lib, it would make sense, but this imho just increases runtime of the test suits and slow down IDE operations :)

Reply:
I can only tell you what I would do, if it is possible to test it on the real results, it would be worth it. But these tests just increases maintenance costs in my eyes.